packages/ethereum/contracts/test/Crypto.t.sol

```diff
@@ -0,0 +1,175 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.6.0 <0.9.0;
+
+import 'forge-std/Test.sol';
+import './utils/Accounts.sol';
+
+import '../src/Crypto.sol';
+
+contract CryptoProxy is HoprCrypto {
+  function modifierIsCurvePoint(CurvePoint memory p) isCurvePoint(p) public {}
+}
+
+contract Crypto is Test, AccountsFixtureTest, HoprCrypto {
+  CryptoProxy crypto;
+  function setUp() public {
+    // use dummy proxy to per-method have gas measurements
+    crypto = new CryptoProxy();
+  }
+
+  function testInvMod(uint256 el) public {
+    el = bound(el, 2 , HoprCrypto.SECP256K1_BASE_FIELD_ORDER - 1);
+
+    uint256 invEl = crypto.invMod(el);
+
+    assertEq(mulmod(el, invEl, HoprCrypto.SECP256K1_BASE_FIELD_ORDER), 1);
+  }
+
+  function testPointToAddress() public {
+    (bool success, bytes memory returnValue) = address(this).staticcall(abi.encodeWithSelector(HoprCrypto.pointToAddress.selector, 0x8318535b54105d4a7aae60c08fc45f9687181b4fdfc625bd1a753fa7397fed75, 0x3547f11ca8696646f2f3acb08e31016afac23e630c5d11f59f61fef57b0d2aa5));
+
+    assertTrue(success);
+    assertEq(address(uint160(uint256(bytes32(returnValue)))), accountA.accountAddr);
+  }
+
+  function testIsCurvePoint() public { 
+    crypto.modifierIsCurvePoint(CurvePoint(0x8318535b54105d4a7aae60c08fc45f9687181b4fdfc625bd1a753fa7397fed75, 0x3547f11ca8696646f2f3acb08e31016afac23e630c5d11f59f61fef57b0d2aa5));
+  }
+
+  function testRevert_NoCurvePoint() public {
+    vm.expectRevert(InvalidCurvePoint.selector);
+    crypto.modifierIsCurvePoint(CurvePoint(0x3547f11ca8696646f2f3acb08e31016afac23e630c5d11f59f61fef57b0d2aa5, 0x8318535b54105d4a7aae60c08fc45f9687181b4fdfc625bd1a753fa7397fed75));
+  }
+
+  function testScalarTimeBasepoint() public {
+    assertEq(HoprCrypto.scalarTimesBasepoint(accountA.privateKey), accountA.accountAddr);
+  }
+
+  function testEcAdd() public {
```

Comment:
Can we maybe add some randomized test, like generate small `n`, compute `n*P` in a for-loop and then check that each step we satisfy a the curve equation?